src/visitors/semantic_analysis_visitor.cpp

```diff
@@ -41,6 +44,17 @@ bool SemanticAnalysisVisitor::check(const ast::Program& node) {
     const auto& sym_table = node.get_symbol_table();
     assert(sym_table != nullptr);
 
+    // this block is going nowhere since nodes below is just its declaration
```

Comment:
This whole block should be removed. I left it just to remind me to mention a confusion I had at the beginning of my implementation where I thought ```get_nodes()``` would return every node where a particular variable appeared.

Reply:
Ok, removed.